server/middleware/middleware.go

```diff
@@ -45,6 +45,10 @@ func (rh *responseHijacker) Write(b []byte) (int, error) {
 	return rh.ResponseWriter.Write(b)
 }
 
+func (rh *responseHijacker) CloseNotify() <-chan bool {
+	return rh.ResponseWriter.(http.CloseNotifier).CloseNotify()
```

Comment:
I don't agree, this middleware explicitly works with martini and [martini.ResponseWriter implements CloseNotifier](https://github.com/cloudwan/gohan/blob/635491f0ac41b34e387a4578689f67bb27a072b9/vendor/github.com/go-martini/martini/response_writer.go#L83-L83). Extensions part isn't coupled with martini and uses standard http.ResponseWriter and that's why we need to check it first.

Additionally, I don't see how we can implement `else` part here. If we decide to implement interface, we should carry it, otherwise otto part wouldn't be able to rely on that.